pandas/core/window/rolling.py

```diff
@@ -400,13 +400,15 @@ def _get_cython_func_type(self, func):
             self._get_roll_func("{}_fixed".format(func)), win=self._get_window()
         )
 
-    def _get_window_indexer(self):
+    def _get_window_indexer(self, index_as_array):
         """
         Return an indexer class that will compute the window start and end bounds
         """
+        if isinstance(self.window, window_indexers.BaseIndexer):
```

Comment:
technically we don’t actually care if this is a BaseIndexer

rather we care that it has a get_windows_bound with the correct signature